handlers.go

```diff
@@ -46,6 +49,93 @@ func httpDefaultError(w http.ResponseWriter, code int) {
 	http.Error(w, http.StatusText(code), code)
 }
 
+// PreloadMetrics for Prometheus for each available repository.
+func (s *Server) PreloadMetrics() error {
+	// No need to preload metrics if those are disabled.
+	if !s.Prometheus {
+		return nil
+	}
+
+	var repoPaths []string
+
+	walkFunc := func(path string, d fs.DirEntry, err error) error {
```

Comment:
The `folderPathValid` function will be used below in line 122, as both the repository's absolute path as well as the relative folder path are needed. I don't see any advantage in using the `splitURLPath` function directly, especially as both `repo.ObjectTypes` and `repo.FileTypes` are being checked in the `walkFunc` to determine if some directory path should be followed further.

As any occurrence of an `repo.ObjectTypes` like directory aborts this path by returning `filepath.SkipDir`, I don't expect too much directory walking. However, there might be special cases where this might perform poorly. I'm open to suggestions how to address this.